### PR TITLE
Fix xmlvalidate tests

### PR DESCRIPTION
**Pull request by djjuhasz**
Fix the xmlvalidate tests so they don't fail in environments that don't have `xmllint` installed.

- Inject an XSD validator into the xmlvalidate activity to allow a variety of implementations
- Use a test XSD validator for tests
- Add a CLI xmlvalidate command for testing
- Add tests for the `xmllint` based XSD validator that only run when the "xmllint" tag is passed to the test runner

### CODE REVIEW

**Review by sevein — COMMENTED** [review]
Many projects deliberately avoid using build tags to gate test categories, as they can obscure build errors and add unnecessary complexity to IDE configurations. Peter Bourgon argued for [using environment variables](https://peter.bourgon.org/blog/2021/04/02/dont-use-build-tags-for-integration-tests.html), which I've implemented in other projects (e.g., sdps_preprocessing). I particularly like this approach because it encourages sticking to the standard `go test ./...` in daily workflows, making it easier to switch between projects without adjusting tools or workflows.

There's no consensus on this, but two other common approaches are:

- Go test flags: these aren't always intuitive or easy to use.
- Build wrappers: for example, Temporal uses a Makefile to create test categories (see [here](https://github.com/temporalio/temporal/blob/11bcaa695d02a6d2d1b3f08262acec4bca1e771a/Makefile#L88-L96)) and runs tests selectively with `make test-*` commands.

One area of consensus is that integration tests should be opt-in, which is why `testing.Short()` is often avoided - something this PR is already adhering to.

**Comment by djjuhasz** [comment]
@sevein okay, I'll switch to using an environment variable.

File: xmlvalidate/activity.go [hunk 1]
```diff
@@ -1,43 +1,55 @@
...
 
 	"go.artefactual.dev/tools/temporal"
 )
 
 const Name = "xml-validate"
 
+// An XSDValidator validates the XML document at xmlPath against the XSD schema
+// at xsdPath.
+type XSDValidator interface {
+	Validate(ctx context.Context, xmlPath, xsdPath string) (string, error)
```
Comment (mcantelon) [inline xmlvalidate/activity.go:16]: Does `xmlPath` need to be specified to be a string?

File: xmlvalidate/activity_test.go [hunk 2]
```diff
@@ -1,126 +1,117 @@
...
 )
 
+type fakeValidator struct {
+	Msg string
+	Err error
+}
+
+var _ xmlvalidate.XSDValidator = &fakeValidator{}
+
+func (v fakeValidator) Validate(ctx context.Context, xmlPath, xsdPath string) (string, error) {
```
Comment (mcantelon) [inline xmlvalidate/activity_test.go:23]: As earlier, does xmlPath need to be specified to be a string?

**Review by mcantelon — APPROVED** [review]
Nice fixes, cleanup, and validator abstraction! Looks good! I had a couple of questions related to one thing (but I might be missing something).

File: xmlvalidate/activity.go (hunk 1 above)
Comment (djjuhasz) [reply xmlvalidate/activity.go:16]: `xmlPath` is specified as a string type, Go allows declaring multiple parameters with the same type using a single type statement: https://go.dev/tour/basics/5 :)

**Pull request merged**